inventory/src/main/java/org/opennms/horizon/inventory/service/LocationGrpcService.java

```
@@ -0,0 +1,127 @@
+
+package org.opennms.horizon.inventory.service;
+
+import java.util.Optional;
+
+import org.mapstruct.factory.Mappers;
+import org.opennms.horizon.inventory.dto.LocationServiceGrpc;
+import org.opennms.horizon.inventory.dto.MonitoringLocationDTO;
+import org.opennms.horizon.inventory.mapper.MonitoringLocationMapper;
+import org.opennms.horizon.inventory.model.MonitoringLocation;
+import org.opennms.horizon.inventory.repository.MonitoringLocationRepository;
+
+import com.google.protobuf.BoolValue;
+import com.google.protobuf.Empty;
+import com.google.protobuf.Int64Value;
+import com.google.rpc.Code;
+import com.google.rpc.Status;
+
+import io.grpc.protobuf.StatusProto;
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import net.devh.boot.grpc.server.service.GrpcService;
+
+@Slf4j
+@GrpcService
+public class LocationGrpcService extends LocationServiceGrpc.LocationServiceImplBase {
+    private final MonitoringLocationRepository locationRepo;
+    private final MonitoringLocationMapper mapper = Mappers.getMapper(MonitoringLocationMapper.class);
+
+    public LocationGrpcService(MonitoringLocationRepository locationRepo) {
+        this.locationRepo = locationRepo;
+    }
+
+    @Override
+    public void createLocation(MonitoringLocationDTO request, StreamObserver<MonitoringLocationDTO> responseObserver) {
+        MonitoringLocation location = locationRepo.findByLocation(request.getLocation());
+        if(location != null) {
+            Status status = Status.newBuilder()
+                .setCode(Code.ALREADY_EXISTS.getNumber())
+                .setMessage("The location already exist")
+                .build();
+            responseObserver.onError(StatusProto.toStatusRuntimeException(status));
+        } else {
+            //if id exist it will update the location
+            responseObserver.onNext(mapper.modelToDTO(locationRepo.save(mapper.dtoToModel(request))));
```

Query on this. If the id exists, should we throw an error? Just for my own knowledge, does the rest of the application do an update in this scenario or throw an error?

---

In classic, the location name is the primary key if I remember correctly. Also, I am unsure if we have a use case to change the location name.